packages/flutter/test/widgets/semantics_tester.dart

```diff
@@ -340,6 +340,54 @@ class SemanticsTester {
     visit(tester.binding.pipelineOwner.semanticsOwner.rootSemanticsNode);
     return result;
   }
+
+  /// Generates an expression that creates a [TestSemantics] reflecting the
+  /// current tree of [SemanticsNode]s.
+  ///
+  /// Use this method to generate code for unit tests.
+  String generateTestSemanticsExpressionForCurrentSemanticsTree() {
+    final SemanticsNode node = tester.binding.pipelineOwner.semanticsOwner.rootSemanticsNode;
+    return _generateSemanticsTestForNode(node, 0);
+  }
+
+  /// Recursively generates [TestSemantics] code for [node] and its children,
+  /// indenting the expression by `indentAmount`.
+  String _generateSemanticsTestForNode(SemanticsNode node, int indentAmount) {
+    final String indent = '  ' * indentAmount;
+    final StringBuffer buf = new StringBuffer();
+    final SemanticsData nodeData = node.getSemanticsData();
+    buf.writeln('new TestSemantics(');
+    if (nodeData.flags != 0)
+      buf.writeln('  flags: ${nodeData.flags},');
```

Comment:
Ah, this is how you got the int values :)

This should probably generate an expression that calculates the int value out of the SemanticsFlag enum values to make the generated code readable.

Reply:
Done.